src/inatinqperf/adaptors/weaviate_adaptor.py

```diff
@@ -1,257 +1,340 @@
-            return
-
-        payload = {
-            "class": self.class_name,
-            "description": "Collection managed by iNatInqPerf",
-            "vectorizer": self.vectorizer,
-            "vectorIndexType": "hnsw",
-            "vectorIndexConfig": {
-                "distance": self._distance_metric,
-                "vectorCacheMaxObjects": 1_000_000,
-            },
-            "properties": [
-                {
-                    "name": "originalId",
-                    "description": "Original integer identifier",
-                    "dataType": ["int"],
-                }
-            ],
-        }
-        response = self._session.post(self._schema_endpoint, json=payload, timeout=self.timeout)
-
-        if response.status_code not in {HTTPStatus.OK, HTTPStatus.CREATED}:
-            # Weaviate returns 422 if class already exists - handle gracefully.
-            if (
-                response.status_code == HTTPStatus.UNPROCESSABLE_ENTITY
-                and "already exists" in response.text.lower()
-            ):
-                logger.info(f"[WeaviateAdaptor] Class {self.class_name} already exists.")
+    def drop_index(self) -> None:
+        """Delete the managed class entirely."""
+        try:
+            self._client.schema.delete_class(self.class_name)
+        except Exception as exc:  # pragma: no cover - defensive programming
+            status_code = getattr(exc, "status_code", None)
+            if status_code == HTTPStatus.NOT_FOUND:
+                logger.info(f"[WeaviateAdaptor] Class {self.class_name} already absent.")
                 return
-            self._raise_error("failed to create class", response)
+            self._handle_exception("failed to delete class", exc)
 
-        logger.info(
-            f"""[WeaviateAdaptor] Created class={self.class_name} """
-            f"""(distance={self._distance_metric} vectorizer={self.vectorizer})"""
-        )
+        logger.info(f"[WeaviateAdaptor] Dropped class={self.class_name}")
 
     def upsert(self, x: Sequence[DataPoint]) -> None:
-        """Insert or update vectors in Weaviate."""
-        ids, vectors = np.empty((len(x),), dtype=np.int64), np.empty((len(x), self.dim), dtype=np.float32)
-        for i, d in enumerate(x):
-            ids[i] = d.id
-            vectors[i] = np.asarray(d.vector, dtype=np.float32)
-
-        if vectors.ndim != self.INATINQ_WEVIATE_QUERY_DIM or vectors.shape[1] != self.dim:
-            msg = "Vectors must be 2-D with shape (n, dim)."
-            raise ValueError(msg)
+        """Insert or update vectors and associated metadata."""
+        datapoints = list(x)
+        if not datapoints:
+            logger.debug("[WeaviateAdaptor] upsert called with empty payload; skipping.")
+            return
 
-        if ids.shape[0] != vectors.shape[0]:
-            msg = "ids and vectors must have matching length"
+        vectors = np.vstack([np.asarray(dp.vector, dtype=np.float32) for dp in datapoints])
```

Comment:
Why are we stacking these? The query vectors should be a sequence of 1D vectors. You are then iterating again below for upsertion, so you might as well iterate over the `datapoints` sequence.